examples/examples.py

```diff
@@ -58,12 +56,27 @@ def stft_plot(signal, sample_rate=SAMPLE_RATE):
 #
 # Envelopes are used to modulate a variety of signals; usually one of pitch, amplitude, or filter cutoff frequency. In
 # this notebook we will use the same envelope to modulate several different audio parameters.
+#
+# ### A note about note-on, note-off behaviour 
+#
+# By default, this envelope reacts as if it was triggered with midi, for example playing a keyboard. Each midi event has a beginning and end: note-on, when you press the key down; and note-off, when you release the key. `sustain_duration` is the amount of time that the key is depressed. During the note-on, the envelope moves through the attack and decay sections of the envelope. This leads to musically-intuitive, but programatically-counterintuitive behaviour.
+#
+# Assume attack is 0.5 seconds, and decay is 0.5 seconds. If a note is held for 0.75 seconds, the envelope won't traverse through the entire attack-and-decay phase (specifically, it will execute the entire attack, and 0.25 seconds of the decay).
+#
+# If this is confusing, don't worry about it. ADSR's do a lot of work behind the scenes to make the playing experience feel natural. Alternately, you may specify one-shot mode (see below), which is more typical of drum machines. 
 
 # Envelope test
 adsr = ADSR(a, d, s, r, alpha)
 envelope = adsr(sustain_duration)
 time_plot(envelope, adsr.sample_rate)
 
+# ### One-Shot Mode
+#
```

Comment:
The fact that you call this a mode suggests it should be a separate param, and not overload the meaning of sustain_duration with a non-smooth (non-differentiable) transition of behavior.

Reply:
^ see above. Again, note-on time is a hyperparameter so I'm not worried about it. Plus even if it were a seperate param it would still be a discrete one, which is not learnable.

Reply:
@maxsolomonhenry main question tho: If I gave you sound and asked you to predict what patch generated it, we'd probably want to include one-shot vs not-one-shot in the parameter space.

If that were the case then learning a boolean parameter would be appropriate. you are right that learning a hard discrete parameter is difficult through gradient descent. but if we relaxed it and turned it into a [0,1] mix parameter between one-shot and not-one-shot mode, then we could. this is one of the cases where the synth becomes a bit more patchable than most humans would want but perhaps a human would want